src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/InvokerUtil.cs

```diff
@@ -150,8 +168,74 @@ internal static InvokeDelegate GenerateInvokeDelegate(MethodInfo method, out int
 
                     return result;
                 };
+            }
+
+            private static InvokeDelegate GenerateInvokeDelegateInternalWithExpressions(int paramCount, bool returnsValue, int[] inputPos, int[] outputPos, MethodInfo method)
+            {
+                var targetParam = Expression.Parameter(typeof(object), "target");
+                var inputsParam = Expression.Parameter(typeof(object[]), "inputs");
+                var outputsParam = Expression.Parameter(typeof(object[]), "outputs");
+                var paramsLocal = Expression.Variable(typeof(object[]), "paramsLocal");
+                var result = Expression.Variable(typeof(object), "result");
+
+                var methodParam = Expression.Constant(method);
+
+                var assignParamsLocal = Expression.Assign(paramsLocal, Expression.Condition(
+                    Expression.GreaterThan(Expression.Constant(paramCount), Expression.Constant(0)),
+                    Expression.NewArrayBounds(typeof(object), Expression.Constant(paramCount)),
+                    Expression.Constant(null, typeof(object[]))));
+
+                var assignParamsLocalValues = inputPos.Length == 0
+                    ? Expression.Block(Expression.Empty())
+                    : Expression.Block(
+                        Enumerable.Range(0, inputPos.Length).Select(i =>
```

Comment:
Nice loop unrolling!!

Reply:
no longer there but inputPos array disappeared too in the rewritten version